assets/src/blocks/Media/MediaEditor.js

```diff
@@ -0,0 +1,155 @@
+
+  return (
+    <Fragment>
+      <Tooltip text={__('Edit text', 'planet4-blocks-backend')}>
+        <header className="articles-title-container">
+          <RichText
+            tagName="h2"
+            className="page-section-header"
+            placeholder={__('Enter title', 'planet4-blocks-backend')}
+            value={video_title}
+            onChange={toAttribute('video_title')}
+            keepPlaceholderOnFocus={true}
+            withoutInteractiveFormatting
+            characterLimit={40}
+            multiline="false"
+          />
+        </header>
+      </Tooltip>
+      <RichText
+        tagName="p"
+        className="page-section-description"
+        placeholder={__('Enter description', 'planet4-blocks-backend')}
+        value={description}
+        onChange={toAttribute('description')}
+        keepPlaceholderOnFocus={true}
+        withoutInteractiveFormatting
+        characterLimit={200}
+      />
+      <MediaPlaceholder
+        labels={{ title: __('Video poster image [Optional]', 'p4ge'), instructions: __('Applicable for .mp4 video URLs only.', 'p4ge')}}
+        icon="format-image"
+        onSelect={ onSelectImage }
+        onSelectURL={ onSelectURL }
+        onError={ onUploadError }
+        accept="image/*"
+        allowedTypes={["image"]}
+      />
+      {
+        (youtube_id && !youtube_id.endsWith('.mp4')) && !embedHtml
+        ? <div className="block-edit-mode-warning components-notice is-error">
+            { __( 'The video URL could not be parsed.', 'planet4-blocks' ) }
+          </div>
+        : <Fragment>
+            {
+              youtube_id && youtube_id.endsWith('.mp4')
+              ? <MediaElementVideo videoURL={ youtube_id } videoPoster={ posterURL } />
+              : <MediaEmbedPreview html={ embedHtml ? embedHtml : null } />
+            }
+          </Fragment>
+      }
+    </Fragment>
+  )
+}
+
+export const MediaEditor = (props) => {
+  const { isSelected, attributes, setAttributes } = props;
+  const { youtube_id, video_poster_img } = attributes;
+  let mediaURL = youtube_id;
+
+  const toAttribute = attributeName => value => setAttributes({ [attributeName]: value });
+
+  // Assume that a non-URL is a YouTube video ID, for back compat.
+  if ( youtube_id && youtube_id.indexOf('/') === -1 ) {
+    mediaURL = `https://www.youtube.com/watch?v=${mediaURL}`;
+  }
+
+  if ( youtube_id && youtube_id.indexOf('youtube.com') > -1 ) {
+    mediaURL.replace('youtube.com', 'youtube-nocookie.com');
+  }
+
+  const { embedHtml, posterURL } = useSelect((select) => {
+    const embedPreview = select('core').getEmbedPreview( mediaURL );
+    const media = select('core').getMedia(video_poster_img)
+    return {
+      embedHtml: embedPreview ? embedPreview.html : null,
+      posterURL: media ? media.media_details.sizes.large.source_url : null
+    }
+  },
+  [ youtube_id, video_poster_img ]);
+
+  useEffect(() => {
+    setAttributes({
+      embedHtml: embedHtml ? embedHtml : null,
```

Comment:
Same as above, these could be shortened to `embedHtml || null` & `posterUrl || null`